app/code/Magento/CatalogInventory/Model/Quote/Item/QuantityValidator/QuoteItemQtyList.php

```diff
@@ -36,7 +36,7 @@ public function getQty($productId, $quoteItemId, $quoteId, $itemQty)
             $this->_checkedQuoteItems[$quoteId][$productId]['items']
         )
         ) {
-            $qty += $this->_checkedQuoteItems[$quoteId][$productId]['qty'];
+            $qty = $this->_checkedQuoteItems[$quoteId][$productId]['qty'];
```

Comment:
I think the goal of this operation is to accumulate all product quantities allocated for all quote items, so in my opinion `+=` is exactly what needs to be used here.

Possibly the cause of the issues is incorrect `quoteItemId` used for some of the calls of this method?